include/mbedtls/config_psa.h

```diff
@@ -848,6 +848,11 @@ extern "C" {
 #define PSA_HAVE_FULL_ECDSA 1
 #endif
 
+#if defined(PSA_WANT_ALG_ECDH) && defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) && \
+    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY)
+#define PSA_HAVE_FULL_ECDH 1
```

Comment:
Right, this indeed might simplify the code. I will do it ;)

Reply:
> supporting ECC private keys without the public part hardly makes sense

Indeed, and [enabling `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR` automatically enables `PSA_WANT_KEY_TYPE_xxx_PUBLIC_KEY`](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.3.0/docs/proposed/psa-conditional-inclusion-c.md#user-content-configuration-symbols-for-key-types).

On the other hand, it's possible to enable `PSA_WANT_ALG_ECDH` without `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR`. This is expected to have no effect on functionality (like enabling `PSA_WANT_ALG_CBC` without any block cipher key type), although an implementation could be excused for adding some dead code, but it would also be correct for the implementation to automatically enable `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR` (because the configuration mechanism never guarantees that something is disabled, for example if the implementation doesn't support fine-grained feature selection, or if the platform itself or some other part of the application requires the feature).

My inclination would be to use `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR && PSA_WANT_ALG_ECDH` around all ECDH code. But I'm ok with defining a separate symbol, though I don't think “full ECDH” is a meaningful name (in fact it's misleading since not all curves need to be supported — in fact with this check it's possible to request zero curves and so not have any actual ECDH functionality).

Reply:
Thanks for the info!

And good point about the name. I checked, and `PSA_HAVE_FULL_ECDH` is only used once, in `check_config.h`. So, I think we should just get rid of it and use `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR && PSA_WANT_ALG_ECDH` instead in that one place. @valeriosetti can you raise a PR for that, and double-check if some other `PSA_HAVE_FULL` macros would deserve a similar treatment?

Reply:
Done #7422